parle.cpp

```diff
@@ -1237,9 +1239,27 @@ _parser_dump(INTERNAL_FUNCTION_PARAMETERS, zend_class_entry *ce) noexcept
 		/* XXX See comment in _lexer_dump(). */
 		// XXX need std:wcout dump!
 #if PARLE_U32
-		zend_throw_exception_ex(ParleParserException_ce, 0, "Parser dump is not supported with UTF-32");
+		using utf_out_iter = lexertl::basic_utf8_out_iterator
+			<const parle::char_type*>;
+		std::basic_stringstream<parle::char_type> ss;
+		parle::string str;
+
+		parsertl::basic_debug<parle::char_type>::dump(par.rules, ss);
+		str = ss.str();
+
+		const parle::char_type* end_str = str.c_str() + str.size();
+		utf_out_iter iter(str.c_str(), end_str);
+		utf_out_iter end(end_str, end_str);
+		std::string u8(iter, end);
+
+		php_write((void*)u8.c_str(), u8.size());
 #else
-		parsertl::debug::dump(par.rules, std::cout);
+		std::stringstream ss;
+		std::string str;
+
+		parsertl::debug::dump(par.rules, ss);
+		str = ss.str();
+		php_write((void*)str.c_str(), str.size());
 #endif
```

Comment:
This seems to be fixing #27, not related to the `readBison` feature. Should it be separated into another PR please? 

Thanks

Reply:
This fix is necessary in order for the `readBison()` test to work.

Thanks

Reply:
I see. Perhaps it just should have been approached in reverse order - first the dump fix, then the bison fix. Then one could split features by PR. Looks like it'd be best to land both in this PR then, to avoid additional inconvenience. Probably should land this one once both features are finalized then. I'll check to add more tests also once this is in.

Thanks

Reply:
Can we merge this one now?

Like you said, we can add extras later.

I'd like to address the other issues ASAP now. I'm hoping they will all be quick from now on.

Reply:
Yeah, it looks fine now. I think the last small piece is to add the missing header to the `package.xml` to make all green, then can continue on the fresh state.

Thanks